upgrade/helpers/openstack.py

```diff
@@ -36,98 +38,105 @@ def get_openstack_client():
+                                'password': password,
+                                'project_name': project_name,
+                                'auth_url': auth_url
+                            }
+                    }
+            }
+    }
+    with open('clouds.yml', 'w') as outfile:
+        yaml.dump(data, outfile, default_flow_style=False)
+    openstack_client = shade.openstack_cloud(cloud='satellite-jenkins')
+    return openstack_client
+
+
+def create_openstack_instance(instance_name, image_name, volume_size,
+                              flavor_name=None, ssh_key=None,
+                              network_name=None):
     """Creates openstack Instance from Image and Assigns a floating IP
     to instance. Also It ensures that instance is ready for testing.
 
-    :param product: A string. A product name of which, instance to create.
     :param instance_name: A string. Openstack Instance name to create.
     :param image_name: A string. Openstack image name from which instance
         to be created.
     :param flavor_name: A string. Openstack flavor_name for instance.
         e.g m1.small.
     :param ssh_key: A string. ssh_key 'name' that required to add
         into this instance.
-    :param int timeout: The polling timeout in minutes to assign IP.
+    :param network_name: A string. Network 'name' that required to create
+        this instance.
 
     ssh_key should be added to openstack project before running automation.
     Else the automation will fail.
 
     The following environment variables affect this command:
 
-    USERNAME
-        The username of an openstack project to login.
-    PASSWORD
-        The password of an openstack project to login.
-    AUTH_URL
-        The authentication url of the project.
-    PROJECT_ID
-        Project ID of an openstack project.
-
+    FLAVOR_NAME
+        Openstack flavor name to create compute settings for instance
+    NETWORK_NAME
+        Name of the network where the instance is created
+    SSH_KEY
+        ssh key to be added into the instance from openstack
     """
-    network_name = 'satellite-jenkins'
-    openstack_client = get_openstack_client()
-    # Validate ssh_key is added into openstack project
-    openstack_client.keypairs.find(name=ssh_key)
-    image = openstack_client.images.find(name=image_name)
-    flavor = openstack_client.flavors.find(name=flavor_name)
-    network = openstack_client.networks.find(label=network_name)
-    floating_ip = openstack_client.floating_ips.create(
-        openstack_client.floating_ip_pools.list()[0].name
-    )
+    env.user = 'root'
+    env.disable_known_hosts = True
+    if flavor_name is None:
+        flavor_name = os.environ.get('FLAVOR_NAME')
+    if network_name is None:
+        network_name = os.environ.get('NETWORK_NAME')
+    if ssh_key is None:
+        ssh_key = os.environ.get('OSP_SSHKEY')
+    openstack_client = shade.openstack_cloud(cloud='satellite-jenkins')
```

Comment:
@jyejare Clouds.yaml , is the default file that the shade object looks at while creating a interface with OSP